pkg/auth/auth.go

```diff
@@ -147,22 +148,69 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		return nil
 	}
 	if unauthorized, ok := err.(docker.ErrUnauthorizedForCredentials); ok {
-		logrus.Debugf("error logging into %q: %v", server, unauthorized)
-		return errors.Errorf("error logging into %q: invalid username/password", server)
+		logrus.Debugf("error logging into %q: %v", key, unauthorized)
+		return errors.Errorf("error logging into %q: invalid username/password", key)
 	}
-	return errors.Wrapf(err, "authenticating creds for %q", server)
+	return errors.Wrapf(err, "authenticating creds for %q", key)
 }
 
-// getRegistryName scrubs and parses the input to get the server name
-func getRegistryName(server string) string {
+// getCredentials returns the auth config for the provided key as well as its registry.
+func getCredentials(systemContext *types.SystemContext, key string, acceptRepositories bool) (auth types.DockerAuthConfig, registry string, err error) {
```

Comment:
It doesn’t make much sense to me to combine the parser and `GetCredentials*` into a single helper: Conceptually it mixes UI and core implementation, and `Logout` doesn’t need to read the pre-existing credentials (at least on the success path).

---

Also, having a separate parser would make it much easier to write unit tests for it (hint).